src/delta_lake/delta_service.py

```diff
@@ -33,6 +35,57 @@
     "vacuum",
 }
 
+DISALLOW_SQL_META_CHARS = {
+    "--",
+    "/*",
+    "*/",
+    ";",
+    "\\",
+}
+
+ALLOWED_STATEMENTS = {
+    "select",
+}
+
+FORBIDDEN_POSTGRESQL_SCHEMAS = {
+    "pg_",
+    "pg_catalog",
+    "information_schema",
+}
+
+
+def _check_query_is_valid(query: str) -> bool:
+    """
+    Check if a query is valid.
+    """
+
+    statements = sqlparse.parse(query)
+    if len(statements) != 1:
+        raise SparkQueryError(f"Query {query} must contain exactly one statement")
+
+    statement = statements[0]
+    if statement.get_type().lower() not in ALLOWED_STATEMENTS:
+        raise SparkQueryError(
+            f"Query {query} must be one of the following: {', '.join(ALLOWED_STATEMENTS)}, got {statement.get_type()}"
+        )
```

Comment:
There could be subqueries / nested queries as well, right? can you traverse through them with this lib?

Reply:
yea. I wasn't able to find an easy way to handle subquery within statement. I added a note regarding to that.

Reply:
Also good enough for now